drift/lib/src/runtime/executor/connection_pool.dart

```diff
@@ -6,71 +8,133 @@ abstract class MultiExecutor extends QueryExecutor {
   ///
   /// Updating statements, or statements that run in a transaction, will be run
   /// with [write]. Select statements outside of a transaction are executed on
-  /// [read].
-  factory MultiExecutor(
-      {required QueryExecutor read, required QueryExecutor write}) {
-    return _MultiExecutorImpl(read, write);
-  }
+  /// [reads].
+  factory MultiExecutor({
+    required List<QueryExecutor> reads,
+    required QueryExecutor write,
+  }) =>
+      _MultiExecutorImpl(reads: reads, write: write);
```

Comment:
Changing the signature like this is a breaking change which I'd like to avoid. Can we keep this factory as it is (use `reads: [read]` when creating the implementation), but create a second one like

```dart
  factory MultiExecutor.withReadPool({
    required List<QueryExecutor> reads,
    required QueryExecutor write,
  }) =>
      _MultiExecutorImpl(reads: reads, write: write);
```